pysmt/solvers/cvc4.py

```diff
@@ -265,6 +262,40 @@ def back(self, expr):
 
         return res
 
+    def __arrays_back(self, arr_expr):
+        '''
+        Helper function to take a CVC4 array and make an Array
+        '''
+        assert arr_expr.getType().isArray(), "Expecting a CVC4 array"
+        visit = [arr_expr]
+        visited = set()
```

Comment:
Good point! I changed the name and also called `visit`, `to_visit` to be a bit more clear. `visited` and the dictionary keys are actually not quite the same. Because it's post-order traversal, `visited` is populated first (in the top-down segment), followed by filling `pysmt_expr`. This could be combined by storing `None` in `pysmt_expr` to mark a node as visited, but I thought this was a little clearer.